Sources/Sentry/SentryFileManager.m

```diff
@@ -20,19 +20,44 @@
 
 NSString *const EnvelopesPathComponent = @"envelopes";
 
+BOOL
+isErrorPathTooLong(NSError *error)
+{
+    NSError *underlyingError = NULL;
+    if (@available(macOS 11.3, iOS 14.5, watchOS 7.4, tvOS 14.5, *)) {
+        underlyingError = error.underlyingErrors.firstObject;
+    }
+    if (underlyingError == NULL) {
+        id errorInUserInfo = [error.userInfo valueForKey:NSUnderlyingErrorKey];
+        if ([errorInUserInfo isKindOfClass:[NSError class]]) {
+            underlyingError = errorInUserInfo;
+        } else {
+            SENTRY_LOG_FATAL(@"Failed to get underlying error from error: %@", error);
```

Comment:
`m`: So for OS versions below `(macOS 11.3, iOS 14.5, watchOS 7.4, tvOS 14.5, *)` we always log this fatal error message here, which can be confusing. Instead, we could simply return NO here and just skip logging a fatal error. It's very unlikely that this happens. To simplify the code we could do an early return above if we can't get the underlyingErrror.

Reply:
Ah yes, missed that one. I will change the code to return early if no underlying error is available